lib/internal/modules/esm/resolve.js

```diff
@@ -359,12 +361,9 @@ function isArrayIndex(key) {
 function resolvePackageTarget(
   packageJSONUrl, target, subpath, packageSubpath, base, internal, conditions) {
   if (typeof target === 'string') {
-    const resolved = resolvePackageTargetString(
+    return finalizeResolution(resolvePackageTargetString(
```

Comment:
Should we have a test to verify this keeps working?

Reply:
There's no case there's a `null` return hence why I could remove this path.